ledger/accountdb_test.go

```diff
@@ -730,6 +744,41 @@ func BenchmarkReadingAllBalances(b *testing.B) {
 	require.Equal(b, b.N, len(bal))
 }
 
+func BenchmarkReadingAllBalancesRAM(b *testing.B) {
+	benchmarkReadingAllBalances(b, true)
+}
+
+func BenchmarkReadingAllBalancesDisk(b *testing.B) {
+	benchmarkReadingAllBalances(b, false)
+}
+
+func benchmarkReadingRandomBalances(b *testing.B, inMemory bool) {
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+	dbs, fn := dbOpenTest(b, inMemory)
+	setDbLogging(b, dbs)
+	defer cleanupTestDb(dbs, fn, inMemory)
+
+	accounts := benchmarkInitBalances(b, dbs, proto)
+
+	qs, err := accountsDbInit(dbs.rdb.Handle, dbs.wdb.Handle)
+	require.NoError(b, err)
+
+	b.ResetTimer()
+	// read all the balances in the database, shuffled by map-iterator hash order
```

Comment:
I'm afraid that it won't be sufficiently random; both the insertion and query are using the same map, so the items would have the same order..